app/src/androidTest/java/com/duckduckgo/app/fire/DatabaseCleanerHelperTest.kt

```diff
@@ -40,20 +41,24 @@ class DatabaseCleanerHelperTest {
     @Before
     fun before() {
         DefaultWebViewDatabaseProvider(context).get()
-
-        // Wait for the database path to be created
-        Thread.sleep(1000)
-
         testee = DatabaseCleanerHelper(coroutineRule.testDispatcherProvider)
     }
 
     @Test
     fun whenCleanDatabaseThenReturnTrue() = runTest {
+        awaitPathCreation()
         assertTrue(testee.cleanDatabase(databaseLocator.getDatabasePath()))
     }
 
     @Test
     fun whenChangeJournalModeToDeleteThenReturnTrue() = runTest {
+        awaitPathCreation()
         assertTrue(testee.changeJournalModeToDelete(databaseLocator.getDatabasePath()))
     }
+
+    private suspend fun awaitPathCreation() {
+        while (databaseLocator.getDatabasePath().isEmpty()) {
```

Comment:
Perhaps we should also guard against this looping forever?

Since we're in a coroutine already, we could use `withTimeout` to put a maximum amount of time that we're willing to wait (before an exception is thrown). This would be a nicer experience as a developer and for the CI system.

Reply:
I've set the timeout to 5s, going to rebuild a few times to check for flakiness...

Reply:
I've re-run this a few times without issue, so hopefully it should be okay now